### ISSUE TITLE
Downloading photo from mobile get the name wrong

### ISSUE THREAD

**Issue by dkm** [cozy/cozy-photos-v2#96]
Downloading a picture when viewing it (clicking on the down arrow) produces
- a file with a very long name (most probably a hash) on my mobile phone (android 4.4.4, AOSP browser). The name is the one encoded in the URL when hovering thumbnails.
- a file with the good name (eg. IMG_1234.JPG) on my regular desktop computer.

**Comment by frankrousseau** [comment]
Ok @dkm thank you for notifying us!

**Comment by m4dz** [comment]
Yep, it's due to the import process of the camera app in Android: when you take a photo from an app (here, you browser), the OS stores it in a temporary folder, with a temporary name. The file is so uploaded with this name in Cozy Photos.

Maybe we could try if a more related name is passed by the OS to the server (but I'm not convinced).

**Comment by dkm** [comment]
sorry i'm not sure we understand each-other... i don't use camera app here, i only try to download a file from the cozy photo app....

**Comment by m4dz** [comment]
Oops, ok. I thought you try to upload a file from your phone to your Cozy Photo :smile:. I'll check that…

**Comment by clochix** [comment]
@m4dz any news ?

**Comment by m4dz** [comment]
Ok, some news, fasten your seat belts! So:
- android:
  - stock browser:
    - download a file with the hash as the filename
    - often use a `jpg` extension even if its another format (such as png)
  - chrome:
    - download a file with correct filename + extension, but doesn't handle correct mimetype in _Documents_ application
    - sometimes download a file without a filename if the data is corrupted in the DB
- desktop:
  - chrome:
    - download a file withe correct filename / extension / mimetype
  - firefox: 
    - download a file with the correct filename / extension if there's no whitespace in filename. With a blank in filename, it truncates the whole string, so `Screen Shot 2015-02-17 at 14.51.05.png` becomes `Screen` (without any extension)
- iOS:
  - go f**\* yourself, you can't _download_ an image in iOS, but it opens it in the browser directly, displaying the hash as filename, and `jpg` as extension, even if it's in another format. Fortunately, Safari mobile doesn't rely on extension to display files, so you can save the image in the camera roll, whatever its format is. The image is so renamed to a camera roll internal compliant name (such as `Photo dd-MM-yyyy HH ii ss.ext` with the correct extension that match the file mimetype)

---

So, we probably should revamp the way a file is disposed to the browser to transfer proper:
- filename, urlencoded to take care of spaces and not-ANSII compatible chars
- extension
- mimetype
- download header

Do I forgot something?

**Comment by frankrousseau** [comment]
Looks good. You can find inspiration in the Files app for that.
